internal/overlord/servstate/handlers.go

```diff
@@ -359,12 +363,16 @@ func (s *serviceData) startInternal() error {
 		_ = s.logs.Close()
 		return fmt.Errorf("cannot start service: %w", err)
 	}
+	logger.Debugf("Service %q started with PID %d", serviceName, s.cmd.Process.Pid)
 	s.resetTimer = time.AfterFunc(s.config.BackoffLimit.Value, func() { logError(s.backoffResetElapsed()) })
 
 	// Start a goroutine to wait for the process to finish.
 	done := make(chan struct{})
 	go func() {
 		waitErr := s.cmd.Wait()
+		if waitErr != nil {
+			logger.Debugf("Service %q Wait() error: %v", serviceName, waitErr)
```

Comment:
Add a debug log here as we didn't log the error message of this waitErr anywhere else.